packages/runtime/container-runtime/src/test/batching.spec.ts

```diff
@@ -0,0 +1,229 @@
+			updateDirtyContainerState: (_dirty: boolean) => {},
+			getLoadedFromVersion: () => undefined,
+			clientId: mockClientId,
+			connected: true,
+		};
+		return mockContext;
+	};
+
+	const mockProvideEntryPoint = async () => ({
+		myProp: "myValue",
+	});
+
+	let containerRuntime: ContainerRuntime;
+	let mockDeltaManager: MockDeltaManager;
+
+	beforeEach(async () => {
+		mockDeltaManager = new MockDeltaManager();
+		containerRuntime = await ContainerRuntime.loadRuntime({
+			context: getMockContext(mockDeltaManager) as IContainerContext,
+			registryEntries: [],
+			existing: false,
+			runtimeOptions: {},
+			provideEntryPoint: mockProvideEntryPoint,
+		});
+	});
+
+	/** Overwrites channelCollection property to make process a no-op */
+	function patchContainerRuntime(cr: ContainerRuntime): ContainerRuntime {
+		const patched = cr as unknown as Omit<ContainerRuntime, "channelCollection"> & {
+			channelCollection: Partial<ChannelCollection>;
+		};
+
+		patched.channelCollection = {
+			process: (
+				message: ISequencedDocumentMessage,
+				local: boolean,
+				localMessageMetadata: unknown,
+			) => {},
+		} satisfies Partial<ChannelCollection>;
+		return patched as unknown as ContainerRuntime;
+	}
+
+	/**
+	 * Returns a batch of messages with the first and last message marked as batch start and end respectively.
+	 */
+	function getBatch(count: number): ISequencedDocumentMessage[] {
+		const messages: ISequencedDocumentMessage[] = [];
+		for (let i = 0; i < count; i++) {
+			messages.push({
+				type: MessageType.Operation,
+				clientId: mockClientId,
+				sequenceNumber: i,
+				minimumSequenceNumber: 1,
+				contents: {
+					type: ContainerMessageType.FluidDataStoreOp,
+					contents: {
+						address: "address",
+						contents: "dummy content",
+					},
+				},
+				clientSequenceNumber: i,
+				timestamp: Date.now(),
+			} as unknown as ISequencedDocumentMessage);
+		}
+		messages[0].metadata = { batch: true };
+		messages[messages.length - 1].metadata = { batch: false };
+		return messages;
+	}
+
+	/**
+	 * Processes the given batch. The batch is processed by pushing each message to the inbound queue and then
+	 * processing the messages in the queue.
+	 */
+	function processBatch(batch: ISequencedDocumentMessage[], cr: ContainerRuntime) {
+		// Push the messages in the inbound queue. This is done because ScheduleManager listens to the "push" event
+		// emitted by the inbound queue to do batch validations.
+		for (const batchMessage of batch) {
+			mockDeltaManager.inbound.push(batchMessage);
+		}
+
+		// Process the messages in the inbound queue.
+		// Process is called on the delta manager because ScheduleManager listens to the "op" event on delta manager
+		// as well to do validation.
+		// Process is called on the container runtime because it is the one that actually processes the messages and
+		// has its own set of validations.
+		let message = mockDeltaManager.inbound.pop();
```

Comment:
Do the tests still work without using the inbound queue?  Seems gratuitous at first glance.

Reply:
`ScheduleManager` does validation on "push" event on the inbound queue. So, we have to push the messages to the queue. I wrote comments to clarify.